packages/ui-test-queries/src/utils/bindElementToMethods.ts

```diff
@@ -24,9 +24,26 @@
 
 export type GenericFunction = (...args: any) => unknown | any
 
-export function bindElementToMethods<K>(element: Element, methods: K) {
+// In an object where keys are functions remove their first argument
+type ObjWithCutFirstArg<T> = {
+  [K in keyof T]: T[K] extends (_: unknown, ...tail: infer TT) => infer R // if a function
+    ? (...args: TT) => R // cut first argument
+    : never // do not allow it, the method will throw an error
+}
```

Comment:
This is the duplicate type I mentioned in my earlier comment.
Also as I see we have a utility type which can be used in this type as well:
```ts
//bindElementToUtilites.ts
type CutFirstArg<F> = F extends (_: any, ...tail: infer TT) => infer R
  ? (...args: TT) => R
  : never
```

So we could reuse this type like this:

```ts
type ObjWithCutFirstArg<T> = {
  [K in keyof T]: CutFirstArg<T[K]>
}
```
What do you think?

Reply:
yes, good idea